lib/linter/linter.js

```diff
@@ -992,91 +1052,121 @@ class Linter {
-         */
-        const filename = path.basename(providedOptions.filename) === "<text>"
-            ? "<text>"
-            : providedOptions.filename;
-
-        // Make options.
-        const options = {
-            ...providedOptions,
-            filename
-        };
+        // Verify.
+        if (processor) {
+            debug("Apply the processor: %o", config.processor);
+            const { preprocess, postprocess, supportsAutofix } = processor;
+            const disableFixes = options.disableFixes || !supportsAutofix;
 
-        // Apply processor.
-        if (config.processor) {
-            const processor = configArray.pluginProcessors.get(config.processor);
+            return this._verifyWithProcessor(
+                textOrSourceCode,
+                config,
+                { ...options, disableFixes, postprocess, preprocess },
+                configArray
+            );
+        }
+        return this._verifyWithoutProcessors(textOrSourceCode, config, options);
+    }
 
-            options.preprocess = processor.preprocess;
-            options.postprocess = processor.postprocess;
-            if (!processor.supportsAutofix) {
+    /**
+     * Verify with a processor.
+     * @param {string|SourceCode} textOrSourceCode The source code.
+     * @param {ConfigData|ExtractedConfig} config The config array.
+     * @param {VerifyOptions&ProcessorOptions} options The options.
+     * @param {ConfigArray} [configForRecursive] The `CofnigArray` object to apply multiple processors recursively.
+     * @returns {LintMessage[]} The found problems.
+     */
+    _verifyWithProcessor(textOrSourceCode, config, options, configForRecursive) {
+        const filename = options.filename || "<input>";
+        const filenameToExpose = normalizeFilename(filename);
+        const text = ensureText(textOrSourceCode);
+        const preprocess = options.preprocess || (rawText => [rawText]);
+        const postprocess = options.postprocess || lodash.flatten;
+        const filterCodeBlock =
+            options.filterCodeBlock ||
+            (blockFilename => blockFilename.endsWith(".js"));
+        const originalExtname = path.extname(filename);
+        const messageLists = preprocess(text, filenameToExpose).map((block, i) => {
+            debug("A code block was found: %o", block.filename || "(unnamed)");
+
+            // Keep the legacy behavior.
+            if (typeof block === "string") {
+                return this._verifyWithoutProcessors(block, config, options);
+            }
+
+            const blockText = block.text;
+            const blockName = path.join(filename, `${i}_${block.filename}`);
 
-                // Use `disableFixes` of https://github.com/eslint/rfcs/tree/master/designs/2018-processors-improvements
-                options.disableFixes = true;
+            // Skip this block if filtered.
+            if (!filterCodeBlock(blockName, blockText)) {
+                debug("This code block was skipped.");
```

Comment:
Can we add blockName into the message? Would make it easier to debug.

Reply:
Actually, the name is shown on the previous line of this message.

![image](https://user-images.githubusercontent.com/1937871/58303446-202ca100-7e2b-11e9-8483-7446289a0aa8.png)